modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/DetectorMappingController.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.web;
+
+import com.expedia.adaptivealerting.modelservice.dao.*;
+import com.expedia.adaptivealerting.modelservice.model.CreateDetectorMappingRequest;
+import com.expedia.adaptivealerting.modelservice.model.DetectorMapping;
+import com.expedia.adaptivealerting.modelservice.model.MatchingDetectorsResponse;
+import com.expedia.adaptivealerting.modelservice.model.SearchMappingsRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api")
+public class DetectorMappingController {
+
+    @Autowired
+    private DetectorMappingService detectorMappingService;
+    
+    @Autowired
+    private RequestValidator requestValidator;
+
+    @RequestMapping(value = "/detector-mapping", produces = "application/json", method = RequestMethod.GET)
+    public DetectorMapping getDetectorMapping(@RequestParam String id) {
+        return detectorMappingService.findDetectorMapping(id);
+    }
+
+    @RequestMapping(value = "/detector-mapping", method = RequestMethod.POST)
+    public ResponseEntity<String> createDetectorMapping(@RequestBody CreateDetectorMappingRequest request) {
+        requestValidator.validateExpression(request.getExpression());
+        requestValidator.validateUser(request.getUser());
+        requestValidator.validateDetector(request.getDetector());
```

Comment:
Consider moving the validation logic to the `CreateDetectorMappingRequest` itself. Either the constructor (preferred) or else a `validate()` method. If you have a lot of different request types that use similar validation logic then you could create an `AbstractDetectorMappingRequest` class with validation helpers.

The advantage is that it cleans up the client code (i.e., here, the controller). Either the client can simply assume the request is valid if the request object does validation in the constructor, or else the client can call a single `request.validate()` without having to know all the details of which properties require validation.

Reply:
Created a validate() function within CreateDetectorMappingRequest